mls-rs-crypto-traits/Cargo.toml

```
@@ -1,6 +1,6 @@
 [package]
 name = "mls-rs-crypto-traits"
-version = "0.9.0"
+version = "0.9.1"
```

```suggestion
version = "0.10.0"
```
Because of `mls-rs-core` that is part of the public API of this crate (e.g. `IntoAnyError`).

---

I think you're right if one wants to be conservative. However, a more fine-grained question would be if `mls-rs-crypto-traits` exports types with breaking changes?

In the case of `IntoAnyError`, I see that it was most recently modified [8 months ago](https://github.com/awslabs/mls-rs/blame/5235fc8cb15295498389f65a9ac6ba175cc392d9/mls-rs-core/src/error.rs#L38). There have been several releases since, so any breaking change in `IntoAnyError` should have been taken into account a long time ago, no?

---

The issue is that as far as rustc is concerned, `mls_rs_core::error::IntoAnyError` from `mls-rs-core 0.17.1` is always different from `mls_rs_core::error::IntoAnyError` from `mls-rs-core 0.18.0` (it does not matter if the definition of the trait hasn't changed). So if I have a project using `mls-rs-core 0.17.1` and `mls-rs-crypto-traits 0.9.0` and I run `cargo update`, my project now uses `mls-rs-core 0.17.1` (no change) and `mls-rs-crypto-traits 0.9.1` which was updated and now uses the definitions from `mls-rs-core 0.18.0` which are considered different from the ones from `mls-rs-core 0.17.1`, and so my code might break.

`IntoAnyError` is just an example of a public trait from `mls-rs-core` that is used in the public API of `mls-rs-crypto-traits`, but any type or trait from `mls-rs-core` used in the public API of `mls-rs-crypto-traits` can become an issue if the version is not updated appropriately. Let's simplify to highlight what the problem is.

Let's consider that `mls-rs-core` defines a type `Foo`, `mls-rs-crypto-traits` defines `fn take_foo(_: Foo)`, and my crate depends on both, instantiates a `Foo` and calls `take_foo`. After running `cargo update`, I get `mls-rs-crypto-traits 0.9.1` which depends on `mls-rs-core 0.18.0`, but my crate still depends on `mls-rs-core 0.17.1` directly (it wasn't updated because it's not considered semver-compatible). So now my crate instantiates `Foo` from `mls-rs-core 0.17.1` and passes it to `take_foo` that expects a `Foo` from `mls-rs-core 0.18.0`, and these 2 types are considered different, so my crate fails to build.

A similar problem can happen with traits. Does that clarify what the problem is?

---

Thankt, that explanation makes it clear! I had not considered this case with a duplication in the transitive dependencies.